lib/utils/join-sql-fragments.ts

```diff
@@ -1,54 +1,55 @@
-'use strict';
-
-function doesNotWantLeadingSpace(str) {
+function doesNotWantLeadingSpace(str: string): boolean {
   return /^[;,)]/.test(str);
 }
-function doesNotWantTrailingSpace(str) {
+function doesNotWantTrailingSpace(str: string): boolean {
   return /\($/.test(str);
 }
 
 /**
  * Joins an array of strings with a single space between them,
  * except for:
- * 
+ *
  * - Strings starting with ';', ',' and ')', which do not get a leading space.
  * - Strings ending with '(', which do not get a trailing space.
- * 
+ *
  * @param {string[]} parts
  * @returns {string}
  * @private
  */
-function singleSpaceJoinHelper(parts) {
-  return parts.reduce(({ skipNextLeadingSpace, result }, part) => {
-    if (skipNextLeadingSpace || doesNotWantLeadingSpace(part)) {
-      result += part.trim();
-    } else {
-      result += ` ${part.trim()}`;
+function singleSpaceJoinHelper(parts: string[]): string {
+  return parts.reduce(
+    ({ skipNextLeadingSpace, result }, part) => {
+      if (skipNextLeadingSpace || doesNotWantLeadingSpace(part)) {
+        result += part.trim();
+      } else {
+        result += ` ${part.trim()}`;
+      }
+      return {
+        skipNextLeadingSpace: doesNotWantTrailingSpace(part),
+        result
+      };
+    },
+    {
+      skipNextLeadingSpace: true,
+      result: ''
     }
-    return {
-      skipNextLeadingSpace: doesNotWantTrailingSpace(part),
-      result
-    };
-  }, {
-    skipNextLeadingSpace: true,
-    result: ''
-  }).result;
+  ).result;
 }
 
 /**
  * Joins an array with a single space, auto trimming when needed.
- * 
+ *
  * Certain elements do not get leading/trailing spaces.
- * 
+ *
  * @param {any[]} array The array to be joined. Falsy values are skipped. If an
  * element is another array, this function will be called recursively on that array.
  * Otherwise, if a non-string, non-falsy value is present, a TypeError will be thrown.
- * 
+ *
  * @returns {string} The joined string.
- * 
+ *
  * @private
  */
-function joinSQLFragments(array) {
+export function joinSQLFragments(array: any[]): string {
```

Comment:
This should be valid (unsure if ts v3.9 will support it):
```ts
type Falsy = 0 | '' | null | undefined | false;
type SQLFragment = string | Falsy | SQLFragment[];
```


w/ the function signature being:
```suggestion
export function joinSQLFragments(array: SQLFragment[]): string {
```

Though this requires a bit of tweaking to get the function working properly (iirc the ts assertions will break a lil - I also converted this file to TS locally and I mostly rewrote the function + optimized it a bit).